src/basicfuns.jl

```diff
@@ -255,10 +260,22 @@ Return `log(exp(x) - 1)` or the “invsoftplus” function.  It is the inverse o
 [`log1pexp`](@ref) (aka “softplus”).
 """
 logexpm1(x::Real) = x <= 18.0 ? log(_expm1(x)) : x <= 33.3 ? x - exp(-x) : oftype(exp(-x), x)
-logexpm1(x::Float32) = x <= 9f0 ? log(expm1(x)) : x <= 16f0 ? x - exp(-x) : oftype(exp(-x), x)
+logexpm1(x::Float32) = x <= 9.0f0 ? log(expm1(x)) : x <= 16.0f0 ? x - exp(-x) : oftype(exp(-x), x)
+
+function softplus(x; a::Real=1.0)
+    if a == 1.0
+        return log1pexp(x)
+    end
+    return log1pexp(a * x) / a
+end
+
+function invsoftplus(y; a::Real=1.0)
+    if a == 1.0
+        return logexpm1(y)
+    end
+    return logexpm1(a * y) / a
+end
```

Comment:
These functions are type unstable (e.g. if `x isa Float32` and `a isa Float64`). I think it would be better to define (and document) something like
```suggestion
softplus(x::Real) = log1pexp(x)
softplus(x::Real, a::Real) = log1pexp(a * x) / a

invsoftplus(y::Real) = logexpm1(y)
invsoftplus(y::Real, a::Real) = logexpm1(a * y) / a
```